### ISSUE TITLE
My parser is not working in production

### ISSUE THREAD

**Issue by mspoulsen** [kach/nearley#445]
Hi,

I am working on a nearley poker parser and everything is working perfectly in development mode. However, as soon as I build and deploy I get this error when trying the parser:

> t.map is not a function
> at Function.r.fromCompiled (nearley.js:173)

I have made a small test input box to show the error:
http://pj-cloud-20190613222249-hostingbucket-dev.s3-website-eu-west-1.amazonaws.com/

If I enter ```A``` into the input box the red border should go away and it does in dev mode. However, in production I get the error mentioned above (check the console).

My code is very straight forward:

```javascript
const nearley = require('nearley')
const holdemGrammar = require('./holdemParser.js')

function createParser(grammar: string) {
  return new nearley.Parser(nearley.Grammar.fromCompiled(grammar))
}

function createHoldemParser() {
  return createParser(holdemGrammar)
}

function parse() {
  const parser = createHoldemParser();

  try {
    this.results = parser.feed(formatSyntax(this.text)).results;
    this.isValid = this.results.length > 0;
  } catch (e) {
    this.isValid = false;
    this.results = e;
  }
}
````

I find this very strange. Can anybody help? :slightly_smiling_face:

**Comment by Vasile-Peste** [comment]
Hello,
why, in the "createParser" function you declared the type of the grammar parameter as string? 

```javascript
Grammar.fromCompiled = function(rules, start) {
    var lexer = rules.Lexer;
    if (rules.ParserStart) {
        start = rules.ParserStart;
        rules = rules.ParserRules;
    }
    var rules = rules.map(function (r) { return (new Rule(r.name, r.symbols, r.postprocess)); });
    var g = new Grammar(rules, start);
    g.lexer = lexer; // nb. storing lexer on Grammar is iffy, but unavoidable
    return g;
 }
```

This is the piece of code you are calling from nearley.js 
If the "map" functions is not defined probably you are not passing an array, just remove the string data type definition, are you using TypeScript?

**Comment by mspoulsen** [comment]
Yeah, you are right. At the time I guess I assumed grammar was a string but it is not the issue here.

I spent quite a long time digging into this. I created a new demo project with just one Nearley component but I was unable to re-create the issue. In my main project I got the error after building.

I then tried to rename the generated js parser to .ts and fix the TS complaints and then it worked! Given that was not a satisfactory long term solution I tried "nearley-loader" for webpack. It works - both in dev and prod mode. So without really finding the cause of the problem I made it go away at least.

Thanks for your suggestion. I will close this issue.
